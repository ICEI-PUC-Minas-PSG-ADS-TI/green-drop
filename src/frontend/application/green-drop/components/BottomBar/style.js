import { StyleSheet } from "react-native";
import { bottomBarColors, shadows } from "@/themes/index";

const getStyles = (theme) => StyleSheet.create({
    container: {
        backgroundColor: theme === 'dark' ? bottomBarColors.dark.background : bottomBarColors.light.background,
        width: '100%',
        height: 60,
        position: 'absolute',
        bottom: 0,
        flexDirection: 'row',
        justifyContent: 'space-between',
        flex: 3,

        // Destaque extra
        ...(
            theme === 'dark'
                ? shadows.dark.top
                : shadows.light.top
        ),

        // Borda sutil
        borderTopWidth: 2,
        borderTopColor: theme === 'dark' ? bottomBarColors.dark.border : bottomBarColors.light.border,
    },
    iconBox: {
        flexDirection: 'column',
        alignItems: 'center',
        flex: 1,
    },
    icon: {
        color: theme === 'dark' ? bottomBarColors.dark.icon : bottomBarColors.light.icon,
    },
    iconText: {
        color: theme === 'dark' ? bottomBarColors.dark.text : bottomBarColors.light.text,
    }
});

export default getStyles;