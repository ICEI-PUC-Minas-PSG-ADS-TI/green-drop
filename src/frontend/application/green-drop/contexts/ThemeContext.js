import React, { createContext, useContext, useState, useEffect } from "react";
import { useColorScheme } from "react-native";
import PropTypes from 'prop-types';

export const ThemeContext = createContext(
  {
  colorScheme: "light",
  toggleTheme: () => {},
}
);

export const ThemeProvider = ({ children }) => {
  const systemColorScheme = useColorScheme();
  const [colorScheme, setColorScheme] = useState(systemColorScheme);
  if (__DEV__) {
    console.log("Theme is:", colorScheme);
  }

  useEffect(() => {
    setColorScheme(systemColorScheme);
  }, [systemColorScheme]);

  const toggleTheme = () => {
    setColorScheme(prev => (prev === 'dark' ? 'light' : 'dark'));
  };

  return (
    <ThemeContext.Provider value={{ colorScheme, toggleTheme }}>
      {children}
    </ThemeContext.Provider>
  );
};
ThemeProvider.propTypes = {
  children: PropTypes.node.isRequired,
};

export const useTheme = () => useContext(ThemeContext);