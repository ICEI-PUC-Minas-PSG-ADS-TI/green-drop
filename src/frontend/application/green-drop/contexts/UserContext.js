/* eslint-disable no-console */
/* global console */
import React, { createContext, useContext, useState, useEffect } from "react";
import PropTypes from "prop-types";
import { usuarioService } from '@/services/Usuario';
import { gameService } from '@/services/Game';

export const UserContext = createContext();

export function UserProvider({ children }) {
  const [user, setUser] = useState(null);
  const [loginStatus, setLoginStatus] = useState(false);
  const [loading, setLoading] = useState(false);
  const [conquistas, setConquistas] = useState([]);
  const [desafios, setDesafios] = useState([]);
  const [historico, setHistorico] = useState([]);
  const [userStats, setUserStats] = useState({ pontos: 0, xp: 0 });

  useEffect(() => {
    (async () => {
      try {
        if(!loginStatus == true) return
        const profile = await usuarioService.getProfile();
        if (profile) {
          setUser(profile);
          setLoginStatus(true);
          await loadGameData();
        } else {
          usuarioService.clearToken();
        }
      // eslint-disable-next-line @typescript-eslint/no-unused-vars
      } catch (e) {
        usuarioService.clearToken();
      }
    })();
  }, [loginStatus]);

  const loadGameData = async () => {
    const [stats, cqs, dsf, hist] = await Promise.all([
      gameService.getUserStats(),
      gameService.getConquistas(),
      gameService.getDesafios(),
      gameService.getHistorico(),
    ]);
    setUserStats(stats);
    setConquistas(cqs);
    setDesafios(dsf);
    setHistorico(hist);
  };

 const loginUsuario = async (email, senha) => { 
  setLoading(true);
  try {
    const data = await usuarioService.login({ email, senha });
    if (!data) throw new Error('Login falhou');
    
    // Busca o perfil atualizado após login
    const profile = await usuarioService.getProfile();
    setUser(profile);
    
    setLoginStatus(true);
    await loadGameData();
    return true;
  } catch (error) {
    console.error("Erro ao fazer login:", error.message);
    return false;
  } finally {
    setLoading(false);
  }
};

  const logoutUsuario = () => {
    usuarioService.clearToken();
    setUser(null);
    setLoginStatus(false);
  };

  const cadastrarUsuario = async ({ nome, email, senha, telefone, imagem }) => {
    setLoading(true);
    try {
      await usuarioService.cadastro({ nome, email, senha, telefone, imagem });
      return await loginUsuario(email, senha);
    } catch (error) {
      console.error("Erro ao cadastrar usuário:", error.message);
      return false;
    } finally {
      setLoading(false);
    }
  };

  return (
    <UserContext.Provider
      value={{
        user,
        userStats,
        conquistas,
        desafios,
        historico,
        loginStatus,
        loading,
        loginUsuario,
        logoutUsuario,
        cadastrarUsuario,
      }}
    >
      {children}
    </UserContext.Provider>
  );
}

UserProvider.propTypes = { children: PropTypes.node };
export const useUserContext = () => useContext(UserContext);
