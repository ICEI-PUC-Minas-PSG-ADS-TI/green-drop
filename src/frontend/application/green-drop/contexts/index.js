export { ThemeProvider, useTheme } from './ThemeContext'
export { UserProvider, useUserContext } from './UserContext'