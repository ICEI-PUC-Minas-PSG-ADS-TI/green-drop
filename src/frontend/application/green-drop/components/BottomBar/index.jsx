import React from "react";
import { View, Text, TouchableOpacity } from "react-native";
import { Alert } from "react-native";
import { useNavigation } from '@react-navigation/native';
import { useTheme } from "@/contexts/ThemeContext";
import { useUserContext } from "@/contexts/UserContext";
import getStyles from "./style";
import { FontAwesomeIcon } from "@fortawesome/react-native-fontawesome";
import { 
  faPlusCircle, 
  faMapMarkerAlt, 
  faTrophy,
  faUser
} from "@fortawesome/free-solid-svg-icons";

export default function BottomBar() {
  const navigation = useNavigation();
  const { colorScheme } = useTheme();
  const styles = getStyles(colorScheme);
  const { loginStatus } = useUserContext();

  // Função para verificar login
  const handleProtectedNavigation = (routeName) => {
    // Verifica se o usuário está logado
    if (!loginStatus) {
      Alert.alert(
        "Login Necessário",
        "Você precisa estar logado para acessar esta funcionalidade",
        [{ text: "OK", onPress: () => navigation.navigate("Login") }]
      );
    } else {
      navigation.navigate(routeName);
    }
  };

  return (
    <View style={styles.container}>
      <TouchableOpacity 
        style={styles.iconBox}
        onPress={() => handleProtectedNavigation('Progresso')}>
        <FontAwesomeIcon icon={faTrophy} size={26} style={styles.icon} />
        <Text style={styles.iconText}>Progresso</Text>
      </TouchableOpacity>
      
      <TouchableOpacity 
        style={styles.iconBox} 
        onPress={() => navigation.navigate('Mapa')}>
        <FontAwesomeIcon icon={faMapMarkerAlt} size={26} style={styles.icon} />
        <Text style={styles.iconText}>Explorar</Text>
      </TouchableOpacity>
      
      <TouchableOpacity 
        style={styles.iconBox} 
        onPress={() => handleProtectedNavigation('Contribuir')}>
        <FontAwesomeIcon icon={faPlusCircle} size={30} style={styles.icon} />
        <Text style={styles.iconText}>Contribuir</Text>
      </TouchableOpacity>
    </View>
  );
}