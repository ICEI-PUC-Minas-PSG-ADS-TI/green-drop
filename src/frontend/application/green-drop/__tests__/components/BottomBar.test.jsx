import React from 'react';
import { render } from '@testing-library/react-native';
import renderer from 'react-test-renderer';
import { NavigationContainer } from '@react-navigation/native';
import { ThemeProvider } from '@/contexts/ThemeContext';
import BottomBar from '@/components/BottomBar/index.jsx';

describe('<BottomBar />', () => {
  it('renderiza sem crashar', () => {
    render(
      <ThemeProvider>
        <NavigationContainer>
          <BottomBar />
        </NavigationContainer>
      </ThemeProvider>
    );
  });

  it('combina com snapshot', () => {
    const tree = renderer.create(
      <ThemeProvider>
        <NavigationContainer>
          <BottomBar />
        </NavigationContainer>
      </ThemeProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
