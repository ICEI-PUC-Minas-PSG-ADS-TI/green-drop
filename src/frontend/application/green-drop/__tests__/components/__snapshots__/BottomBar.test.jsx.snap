// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BottomBar /> combina com snapshot 1`] = `null`;
